docs/release-process.md

```diff
@@ -0,0 +1,175 @@
+# samba-operator Release Process
+
+## Preparation
+
+The samba-operator project has a dedicated branch, called `release`, for
+release versions. This is done to update files, in particular the tags of the
+default containers in `internal/conf`, that control dependencies. Tags are
+applied directly to this branch and only this branch.
+
+
+Prior to tagging, we must update the `release` branch to "contain" all the
+latest changes from the `master` branch. We do this by merging `master` into
+`release`.
+Example:
+
+```
+git checkout master
+git pull --ff-only
+git checkout release
+git pull --ff-only
+git merge master
+# resolve any conflicts
+```
+
+Now we need to update the versions of the containers that the operator deploys
+by default. Edit `internal/conf/config.go` to update the versions of
+the samba-server, samba-metrics, and svcwatch containers if there are new
+released versions for these images.
+
+> [!IMPORTANT]
+> The projects that the samba-operator includes/depends on must be released
```

Comment:
we should name/link those projects in the text IMO.

Reply:
ok, done.